### PR TITLE
short hands

### PR DESCRIPTION
**Pull request by smyrman**
- lint: work-around linter warning
- add several new new checks and short-hands

### CODE REVIEW

**Comment by smyrman** [comment]
This appear to have been laying around for a while...

File: check_middleware.go [hunk 1]
```diff
@@ -52,3 +54,13 @@ func OnIndex(i int, c Check) CheckFunc {
 		return nil
 	}
 }
+
+// Iterate runs the firest check on index 0, the second on index 1 etc, and
```
Comment (bbergshaven) [inline check_middleware.go:58]: small typo: firest => first

File: subjson/check.go [hunk 2]
```diff
@@ -1,9 +1,33 @@
...
 import (
+	"encoding/json"
+
 	"github.com/clarify/subtest"
 )
 
+// Fields is a short-hand for OnMap(subtest.Fields{...})
+type Fields subtest.Fields
+
+// Check validates the json decodd vf against m, expecting vf to return a json
```
Comment (bbergshaven) [inline subjson/check.go:12]: decodd => decoded

File: subjson/check.go [hunk 3]
```diff
@@ -1,9 +1,33 @@
...
+// map.
+func (m Fields) Check(vf subtest.ValueFunc) error {
+	return OnMap(subtest.Fields(m)).Check(vf)
+}
+
+// RawEqual is a short-hand for subtest.DeepEqual(json.RawMessage(s))
+type RawEqual []byte
+
+// Check validates the arw json equals m.
```
Comment (bbergshaven) [inline subjson/check.go:21]: arw => raw

**Comment by smyrman** [comment]
Fixed typos.

**Review by bbergshaven — APPROVED** [review]

**Pull request merged**